sdk/rust/oak_runtime_compat/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright 2019 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Functionality to allow use of Rust code within the Oak Runtime.
+
+#![no_std]
+
+#![feature(lang_items)]
+#![feature(allocator_api)]
+#![feature(alloc_prelude)]
+#![feature(alloc_error_handler)]
+#![feature(fn_traits)]
+#![feature(rustc_private)]
+#![feature(never_type)]
```

Comment:
Interesting, does `!` not work otherwise? I don't remember having to manually add this feature before when I used it.

Reply:
It's required for building with `cargo build`, but not for the Bazel version. Some of the changes were to make it `cargo build` compatible

Reply:
My understanding was that you need this feature only if you want to use `!` as a first class citizen type, in positions other than as return from a fn. Otherwise does it mean that even things like https://doc.rust-lang.org/std/process/fn.exit.html cannot be expressed in stable rust (which does not have unstable features)?

Reply:
Well `std::process::exit` inherits the flag from https://github.com/rust-lang/rust/blob/6b561b4917e803c4be4ca44d8e552b680cb9e380/src/libstd/lib.rs#L284, I'd remove the flag but I can't get the cargo version to compile without it.

Reply:
(Comment above is "std is implemented with unstable features, many of which are internal compiler details that will never be stable")